user-interface/0002-WCON-connect_vega_wallet.md

```diff
@@ -0,0 +1,76 @@
+  
+  - if the dapp is unable to connect for technical reason (e.g. CORS): **must** see an explanation of the error, and a method of fixing the issue  <a name="0002-WCON-016" href="#0002-WCON-016">0002-WCON-016</a>
+  
+
+- ~~Browser wallet~~ `not available yet`
+  
+- Fairground hosted wallet
+  - **must** input a wallet name <a name="0002-WCON-017" href="#0002-WCON-017">0002-WCON-017</a>
+  - **must** input a password <a name="0002-WCON-018" href="#0002-WCON-018">0002-WCON-018</a>
+  - if success: **must** see that the wallet is connected and details of connected key <a name="0002-WCON-019" href="#0002-WCON-019">0002-WCON-019</a>
+  - if failure: **must** see reason for failure <a name="0002-WCON-020" href="#0002-WCON-020">0002-WCON-020</a>
+  - *note: the fairground hosted wallet is configured to automatically approve connections from dapps so there is no need to key selection.*
+  
+- **must** have the option to select a different method / wallet type if I change my mind <a name="0002-WCON-021" href="#0002-WCON-021">0002-WCON-021</a>
+
+... so I can use the interface read data about my key/party or request my wallet to broadcast transactions to a Vega network.
+
+## Disconnect wallet
+
+When wishing to disconnect my wallet, I...
+
+- **must** see an option to disconnect wallet <a name="0002-WCON-022" href="#0002-WCON-022">0002-WCON-022</a>
+- **should** see confirmation that wallet has been disconnected <a name="0002-WCON-023" href="#0002-WCON-023">0002-WCON-023</a>
+- **should** see prompt to connect a wallet, after disconnect <a name="0002-WCON-024" href="#0002-WCON-024">0002-WCON-024</a>
+
+... so that I can protect my wallet from malicious use or select a different wallet to connect to
+
+
+## Select and switch keys
+
+when looking to do something with a specific key (or set of keys) from my wallet, I...
+
+- **must** see what key is currently selected (if any) <a name="0002-WCON-025" href="#0002-WCON-025">0002-WCON-025</a>
+- **must** see an option to switch keys, and a list of keys that are approved from the connected wallet <a name="0002-WCON-026" href="#0002-WCON-026">0002-WCON-026</a>
+
+- for each key:
+  - **must** see the first and last 6 digits of the [public key](7001-DATA-data_display.md#public-keys). <a name="0002-WCON-027" href="#0002-WCON-027">0002-WCON-027</a>
+  - **should** be able to see the whole public key <a name="0002-WCON-028" href="#0002-WCON-028">0002-WCON-028</a>
+  - **must** be able to copy to clipboard the whole public key <a name="0002-WCON-029" href="#0002-WCON-029">0002-WCON-029</a>
+  - **must** see the key name/alias (meta data) <a name="0002-WCON-030" href="#0002-WCON-030">0002-WCON-030</a>
+  - **should** see what non-zero assets that kay has <a name="0002-WCON-031" href="#0002-WCON-031">0002-WCON-031</a>
+  - **could** see the Total asst balances (inc associated) <a name="0002-WCON-032" href="#0002-WCON-032">0002-WCON-032</a>
```

Comment:
```suggestion
  - **could** see the total asset balances (including associated) <a name="0002-WCON-032" href="#0002-WCON-032">0002-WCON-032</a>
```

Reply:
what about nominated & locked for VEGA tokens?

Reply:
resolved in: https://github.com/vegaprotocol/specs/pull/1181/commits/ca3dddbb8f6d10c1e3282176ff1c9d598e9b6c81